File: src/main/java/org/jukeboxmc/block/BlockSmallAmethystBud.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemSmallAmethystBud;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockSmallAmethystBud extends Block{

    public BlockSmallAmethystBud() {
        super( "minecraft:small_amethyst_bud" );
    }

    @Override
    public ItemSmallAmethystBud toItem() {
        return new ItemSmallAmethystBud();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.SMALL_AMETHYST_BUD;
    }

    @Override
    public boolean isTransparent() {
        return true;
    }

    @Override
    public boolean isSolid() {
        return false;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockSmallAmethystBud.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.SmallAmethystBud
import org.jukeboxmc.api.block.data.BlockFace
import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.item.JukeboxItem
import org.jukeboxmc.server.player.JukeboxPlayer
import org.jukeboxmc.server.world.JukeboxWorld

class BlockSmallAmethystBud(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), SmallAmethystBud {

    override fun placeBlock(
        player: JukeboxPlayer,
        world: JukeboxWorld,
        blockPosition: Vector,
        placePosition: Vector,
        clickedPosition: Vector,
        itemInHand: JukeboxItem,
        blockFace: BlockFace
    ): Boolean {
        val blockDown = this.getRelative(BlockFace.DOWN)
        if (!blockDown.isSolid()) return false
        this.setBlockFace(blockFace)
        return super.placeBlock(player, world, blockPosition, placePosition, clickedPosition, itemInHand, blockFace)
    }

   override fun getBlockFace(): BlockFace {
       return BlockFace.valueOf(this.getStringState("minecraft:block_face"))
   }

   override fun setBlockFace(value: BlockFace): SmallAmethystBud {
       return this.setState("minecraft:block_face", value.name.lowercase())
   }
}